src/core/function_class.py

```diff
@@ -108,3 +108,34 @@ def __next__(self) -> tuple[Tensor, Tensor]:
 
     def evaluate(self, x_batch: Tensor, *params: Tensor) -> Tensor:
         return self._in_fc.evaluate(x_batch, *params)
+
+class CombinationFunctionClasses(FunctionClass):
+    
+    def __init__(self, inner_function_classes: Iterable[FunctionClass]):
+        self._in_fcs = inner_function_classes
+        
+        self.p_dists=[fc.p_dist for fc in self._in_fcs]
+
+        for num, function_class in enumerate(self._in_fcs): #Verify that the necessary paramaters are the same for all the inner function classes
+            if num==0:
+                self.x_dist=function_class.x_dist
+                self.batch_size=function_class.batch_size
+                self.sequence_length=function_class.sequence_length
+                self.x_dim=function_class.x_dim
+                self.x_curriculum_dim = function_class.x_curriculum_dim
+                self.y_dim = function_class.y_dim
+                
+            else:
+                assert self.x_dist==function_class.x_dist
+                assert self.x_curriculum_dim==function_class.x_curriculum_dim
+                assert self.y_dim==function_class.y_dim
+
+
+    def _init_param_dist(self) -> Distribution:
+        """Produce the distribution with which to sample parameters"""
+        raise NotImplementedError(f"Abstract class CombinationFunctionClasses does not have a parameter distribution!")
+
+
+    def evaluate(self, x_batch: Tensor, *params: Tensor) -> Tensor:
+        """Produce a Tensor of shape (batch_size, sequence_length, y_dim) given a Tensor of shape (batch_size, sequence_length, x_dim)"""
+        raise NotImplementedError(f"Abstract class CombinationFunctionClasses does not implement `.evaluate(xs)`!")
```

Comment:
should this be in wrappers.py in function_classes as opposed to core?

Reply:
Maybe, I just saw that the modified one was there as well. But I can move it if you want